src/fastcs_odin/frame_processor.py

```diff
@@ -99,6 +104,46 @@ class FrameProcessorAdapterController(OdinDataAdapterController):
     _subcontroller_label = "FP"
     _subcontroller_cls = FrameProcessorController
 
+    def _collect_commands(
+        self,
+        path_filter: Sequence[str | tuple[str, ...] | re.Pattern],
+        command_name: str,
+    ):
+        commands = []
+
+        controllers = list(_filter_sub_controllers(self, path_filter))
+
+        for controller in controllers:
+            try:
+                cmd = getattr(controller, command_name)
+                commands.append(cmd)
+            except AttributeError as err:
+                raise AttributeError(
+                    f"Sub controller {controller} "
+                    f"does not have command '{command_name}'."
+                ) from err
```

Comment:
It would be really useful to provide the parent controller and the path filter as well as the leaf sub controller her, so that it is clear why we are expecting this sub controller needs to have this attribute.